src/plugins/intel_cpu/src/nodes/executors/acl/acl_common_executor.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "acl_common_executor.hpp"
+#include "acl_utils.hpp"
+#include "nodes/executors/memory_arguments.hpp"
+#include "utils/debug_capabilities.h"
+
+namespace ov {
+namespace intel_cpu {
+
+static ACLMemoryInfo initTensorInfo(const MemoryPtr& memoryPtr,  const ACLTensorAttrs& attrs) {
+    auto acl_tensor_type   = precisionToAclDataType(memoryPtr->getPrecision());
+    auto acl_tensor_layout = getAclDataLayoutByMemoryDesc(memoryPtr->getDescPtr());
+
+    ACLMemoryInfo aclMemoryInfo = nullptr;
+    if (acl_tensor_type != arm_compute::DataType::UNKNOWN) {
+        auto collapsed_dims = collapse_dims_to_max_rank(memoryPtr->getStaticDims(), attrs.maxDimsShape);
+        auto acl_tensor_shape = shapeCast(collapsed_dims);
+        if (attrs.hasLayoutTypeNHWC) {
+            changeLayoutToNH_C({&acl_tensor_shape});
+        }
+        aclMemoryInfo = std::make_shared<arm_compute::TensorInfo>(
+                acl_tensor_shape, 1,
+                acl_tensor_type,
+                acl_tensor_layout);
+    }
+    return aclMemoryInfo;
+}
+
+static ACLMemory initTensor(const ACLMemoryInfo& aclMemoryInfo) {
+    ACLMemory aclMemory = nullptr;
+    if (aclMemoryInfo) {
+        aclMemory = std::make_shared<arm_compute::Tensor>();
+        aclMemory->allocator()->init(*aclMemoryInfo);
+    }
+    return aclMemory;
+}
+
+arm_compute::ITensorInfo* ACLCommonExecutor::getACLInfo(const ACLMemory& aclMemory) {
+    if (aclMemory) {
+        return aclMemory->info();
+    }
+    return nullptr;
+}
+
+bool ACLCommonExecutor::update(const MemoryArgs &memory) {
+    for (auto& cpu_mem_ptr : memory) {
+        // Initialize arm_compute::TensorInfo object
+        auto aclTensorInfo = initTensorInfo(cpu_mem_ptr.second, aclTensorAttrs);
```

Comment:
What about to restructure the code flow, so:
1) Create tensor info and validate.
So, we don't create some default tensor info and update it. Instead, we should create a final version of tensor info right away.
To avoid any overhead, probably, it would be better to just store every tensor info as a separate variable and to not try to put it into some container
2) Create Tensor using tensor info
3) Configure

Reply:
@EgorDuplensky corrected